pkgs/development/ocaml-modules/junit/default.nix

```diff
@@ -0,0 +1,17 @@
+{ lib, fetchFromGitHub, buildDunePackage, ptime, tyxml }:
+
+let
+  shared = (import ./shared.nix) { inherit lib fetchFromGitHub; };
+in
```

Comment:
You can avoid some boilerplate by moving the contents of `shared` into the `junit` derivation. Then the other packages can inherit from thuir `junit` dependency rather than from `shared`.

Reply:
Made some changes. But I'm not sure if it fits what you mean.

Reply:
You can just remove the `shared.nix` file.

Reply:
Then replicate all the data?

Reply:
I don't want to replicate the data because on updates I'd have to touch several files.

Reply:
The data is written once in `default.nix` in the `junit` derivation. Both other packages depend on `junit` and can inherit the data from here.

Reply:
I understand you want junit_alcotest, junit_ounit to inherit data data from junit (default). But I'm not sure of implementation method for this. This might be obvious to you but I'm not there yet. Would be nice if you provided some example of how to do this.

Reply:
It is exactly what you had before, except that instead of `inherit (shared) src version…` you have `inherit (junit) src version …`.

Reply:
Done.